src/ChronicleSession.js

```diff
@@ -0,0 +1,69 @@
+// @flow
+import { config, S3 } from 'aws-sdk';
+import { ParquetWriter, ParquetSchema } from 'parquetjs'
+import { createReadStream } from 'fs';
+
+const PATH = '/tmp';
+const BUCKET = 'appdev-register';
+
+class ChronicleSession {
+  app: string;
+  cacheSize: number;
+  logMap: Map<string, Object[]>;
+  s3: S3;
+
+  constructor(
+    accessKey: string, 
+    secretKey: string, 
+    app: string, 
+    cacheSize: number = 10,
+  ) {
+    this.app = app;
+    this.logMap = new Map();
+    this.cacheSize = cacheSize;
+
+    config.update({
+      accessKeyId: accessKey,
+      secretAccessKey: secretKey,
+    })
+    this.s3 = new S3();
+  }
+
+  async writeLogs(eventName: string, eventType: ParquetSchema) {
+    const filename = `${Date.now()}.parquet`;
+    const schema = new ParquetSchema(eventType);
+    const writer = await ParquetWriter.openFile(schema, `${PATH}/${filename}`);
+
+    const logs = this.logMap.get(eventName);
+    if (logs === undefined) { // sanity check
+      return;
+    }
+
+    logs.forEach((log) => { writer.appendRow(log) });
```

Comment:
`log => writer.appendRow(log)`?

Reply:
writer.appendRow(log) is spec from the ParquetWriter to add a row in the parquet file

Reply:
:+1: , i just prefer the shorthand for the one-argument, one-line arrow function